### PR TITLE
feat(project): partial project update

### PR DESCRIPTION
**Pull request by lassemand**

### CODE REVIEW

**Comment by lassemand** [comment]
With the current setup partial project update can not be tested.

My suggestion is that a separate instance of api-auth is spinned up behind ambassador prior to the tests on project update is running. That would no matter what be dealt with in another PR

File: packages/stable/src/types.ts [hunk 1]
```diff
@@ -209,9 +209,17 @@ export interface ApiKeyRequest {
   serviceAccountId: CogniteInternalId;
 }
 
+export type ArrayPatchString =
+  | { set: Set<string> }
+  | { add?: Set<string>; remove?: Set<string> };
```
Comment (polomani) [inline packages/stable/src/types.ts]: Shouldn't it be just an array here?

File: packages/stable/src/types.ts [hunk 2]
```diff
@@ -209,9 +209,17 @@ export interface ApiKeyRequest {
...
+
 export type ArrayPatchLong =
-  | { set: number[] }
-  | { add?: number[]; remove?: number[] };
+  | { set: Array<number> }
+  | { add?: Array<number>; remove?: Array<number> };
+
+export type ArrayPatchClaimNames =
+  | { set: Array<ClaimName> }
+  | { add?: Array<ClaimName>; remove?: Array<ClaimName> };
```
Comment (polomani) [inline packages/stable/src/types.ts]:
```suggestion
  | { add?: ClaimName[]; remove?: ClaimName[] };
```
we use short syntax in other places

File: packages/stable/src/types.ts (hunk 1 above)
Comment (lassemand) [reply packages/stable/src/types.ts]: Nope. They need to be unique.

Comment (polomani) [reply packages/stable/src/types.ts]: have you checked that this actually works?
```
JSON.stringify(new Set("a", "b"))
```
prints  `{}`

Comment (polomani) [reply packages/stable/src/types.ts]: tbh, that's up to the api to perform such validations.
sets are not necessarily easy to work with

Comment (lassemand) [reply packages/stable/src/types.ts]: That seems like an unfortunate behaviour.

**Review by polomani — APPROVED** [review]
LGTM

**Comment by lassemand** [comment]
@vegardok

**Review by begozcan — APPROVED** [review]

**Pull request merged**